core/base/supportedbases.go

```diff
@@ -4,65 +4,25 @@
 package base
 
 import (
-	"sort"
-	"time"
-
-	"github.com/juju/collections/set"
 	"github.com/juju/errors"
 )
 
 // ControllerBases returns the supported workload bases available to it at the
 // execution time.
-func ControllerBases(now time.Time, requestedBase Base, imageStream string) ([]Base, error) {
-	return seriesToBase(requestedBase, func(requestedSeries string) (set.Strings, error) {
-		// The DistroInfo is currently in series, so we need to convert it to
-		// base to get the correct series.
-		return ControllerSeries(now, requestedSeries, imageStream)
-	})
+func ControllerBases() []Base {
+	return []Base{
+		MakeDefaultBase("ubuntu", "22.04"),
+		MakeDefaultBase("ubuntu", "20.04"),
+	}
 }
 
 // WorkloadBases returns the supported workload bases available to it at the
 // execution time.
-func WorkloadBases(now time.Time, requestedBase Base, imageStream string) ([]Base, error) {
-	return seriesToBase(requestedBase, func(requestedSeries string) (set.Strings, error) {
-		// The DistroInfo is currently in series, so we need to convert it to
-		// base to get the correct series.
-		return WorkloadSeries(now, requestedSeries, imageStream)
-	})
-}
-
-func seriesToBase(requestedBase Base, fn func(string) (set.Strings, error)) ([]Base, error) {
-	var requestedSeries string
-	if !requestedBase.Empty() {
-		var err error
-		requestedSeries, err = GetSeriesFromBase(requestedBase)
-		if err != nil {
-			return nil, errors.Annotatef(err, "%q is not a valid base", requestedBase.DisplayString())
-		}
-	}
-
-	series, err := fn(requestedSeries)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-
-	bases := map[Base]struct{}{}
-	for _, s := range series.Values() {
-		b, err := GetBaseFromSeries(s)
-		if err != nil {
-			return nil, errors.Trace(err)
-		}
-		bases[b] = struct{}{}
-	}
-
-	results := make([]Base, 0, len(bases))
-	for b := range bases {
-		results = append(results, b)
+func WorkloadBases() []Base {
```

Comment:
Did we want to include the relevant non LTS bases here as well (for workloads).
There's merit in just restricting to LTS bases. But till now, we've allowed people to deploy to supported non-LTS series. With reading distro info, as series transitioned to out f support, they would drop off. But now we're hard coding, it's probably safer to go with LTS and people can still opt in to deploying to 23.10 or whatever. If we're going down that road, then we don't need separate Controller and Workload bases methods.

Reply:
Yes I think you're right, juju itself shouldn't limit workloads to LTS, we should limit it to beyond bionic (or is it focal and beyond?)

Reply:
@wallyworld We're trying to move towards removing force, not adding more use of it. I also have concerns about hardcoding the non LTS release.  20.10 and 21.04 are EOS maybe EOL.